lib/puppet/http/retry_after_handler.rb

```diff
@@ -0,0 +1,47 @@
+require 'date'
+require 'time'
+
+class Puppet::HTTP::RetryAfterHandler
+  def initialize(retry_limit, max_sleep)
+    @retry_limit = retry_limit
+    @max_sleep = max_sleep
+  end
+
+  def retry_after?(request, response)
+    case response.code
+    when 429, 503
```

Comment:
should 408(HTTPRequestTimeout)  and 504(HTTPGatewayTimeout) be also handled?

Reply:
Yeah, would be good to add that eventually, though for right now I'm going for parity with the current implementation: https://github.com/puppetlabs/puppet/blob/ba820a071ea3367fb1da7906eba222d1707565e9/lib/puppet/network/http/connection.rb#L215. I'll file a separate ticket for those codes.

Reply:
https://tickets.puppetlabs.com/browse/PUP-10137